src/Promotion/Decorator/OrderPromotionsUsageModifierDecorator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Advanced Promotion plugin for Sylius.
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusAdvancedPromotionPlugin\Promotion\Decorator;
+
+use MonsieurBiz\SyliusAdvancedPromotionPlugin\Entity\PromotionCouponsAwareInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Promotion\Modifier\OrderPromotionsUsageModifier;
+use Sylius\Component\Core\Promotion\Modifier\OrderPromotionsUsageModifierInterface;
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
+use Symfony\Component\DependencyInjection\Attribute\AutowireDecorated;
+
+#[AsDecorator('sylius.promotion_usage_modifier')]
+final class OrderPromotionsUsageModifierDecorator implements OrderPromotionsUsageModifierInterface
+{
+    public function __construct(
+        #[AutowireDecorated]
+        private readonly OrderPromotionsUsageModifier $promotionProcessor,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏you should use the interface, if we have several decorators you could have a problem:

```suggestion
        private readonly OrderPromotionsUsageModifierInterface $promotionProcessor,
```